### PR TITLE
Allow mongodb+srv:// protocol

### PR DESCRIPTION
**Pull request by malarzm**
This should fix https://github.com/doctrine/DoctrineMongoDBBundle/issues/452

### CODE REVIEW

**Comment by jmikola** [comment]
For reference, `mongodb+srv://` is documented in [initial seedlist discovery specification](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst), which is implemented in `ext-mongodb` 1.4.0.

File: tests/Alcaeus/MongoDbAdapter/Mongo/MongoClientTest.php [hunk 1]
```diff
@@ -9,6 +9,24 @@
...
+        $this->assertSame($expected, (string) (new \MongoClient($uri, ['connect' => false])));
+    }
+
+    public function provideConnectionUri()
+    {
+        yield ['default', sprintf('mongodb://%s:%d', \MongoClient::DEFAULT_HOST, \MongoClient::DEFAULT_PORT)];
+        yield ['localhost', 'mongodb://localhost'];
+        yield ['mongodb://localhost', 'mongodb://localhost'];
+        if (version_compare(phpversion('mongodb'), '1.4.0', '>=')) {
+            yield ['mongodb+srv://localhost', 'mongodb+srv://localhost'];
```
Comment (jmikola) [inline tests/Alcaeus/MongoDbAdapter/Mongo/MongoClientTest.php]: libmongoc requires that there by only a single hostname and that the hostname string contain at least two dots (https://github.com/mongodb/mongo-c-driver/blob/30085af2124de68fd5ad714309f3ea2688489dee/src/mongoc/mongoc-uri.c#L430). I'm not sure if requiring two dots is correct, as the [specification](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst#specification) only suggests a hostname and domain suffix. I'll follow up if this turns out to be a libmongoc bug, but let's use `foo.example.com` for now.

Comment (jmikola) [reply tests/Alcaeus/MongoDbAdapter/Mongo/MongoClientTest.php]: Ah, this actually isn't a bug. I misread the spec. It requires `{hostname}.{domainname}`, where a domain name includes its own dot before the TLD suffix.

There's also a test for this in [not-enough-parts.json](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/tests/not-enough-parts.json).

Comment (malarzm) [reply tests/Alcaeus/MongoDbAdapter/Mongo/MongoClientTest.php]: Thanks a lot for finding this!

**Comment by alcaeus** [comment]
@malarzm Two builds failed: [the first](https://travis-ci.org/alcaeus/mongo-php-adapter/jobs/348603425) is the test suite sanity check which runs the test suite against `ext-mongo` to ensure the behavior expected in tests is the same behavior shown by `ext-mongo`. Since we're "only" talking URL parsing here, I'll let you decide whether you want to fix it or whether you want to skip it using `skipTestIf(extension_loaded('mongo'));`.

The [second build that failed](https://travis-ci.org/alcaeus/mongo-php-adapter/jobs/348603428) seems to have an issue with your data provider. I suppose the lowest version doesn't like your data provider. I'll try bumping phpunit dependency in a separate PR, should be done shortly.

**Comment by alcaeus** [comment]
Thanks @malarzm!

**Comment by miljojohn** [comment]
Failed to parse MongoDB URI: 'mongodb+srv://testuser:passwordfotheuser@cluster0-3xdfs.mongodb.net'. Invalid URI Schema, expecting 'mongodb://'."

**Comment by alcaeus** [comment]
Which versions of `mongodb/mongodb` and `ext-mongodb` do you have installed?

**Comment by miljojohn** [comment]
- Installing mongodb/mongodb (1.2.0): Loading from cache
 - Installing alcaeus/mongo-php-adapter (1.1.5): Loading from cache

**Comment by alcaeus** [comment]
Please post the version of ext-mongodb you’re using. To get you started, you need at least version 1.4.0 of the driver to be using the mongodb+srv protocol, as written above.

**Comment by miljojohn** [comment]
MongoDB support => enabled
MongoDB extension version => 1.4.2
MongoDB extension stability => stable
libbson bundled version => 1.9.3
libmongoc bundled version => 1.9.3
libmongoc SSL => enabled
libmongoc SSL library => OpenSSL
libmongoc crypto => enabled
libmongoc crypto library => libcrypto
libmongoc crypto system profile => disabled
libmongoc SASL => enabled
libmongoc compression => enabled
libmongoc compression snappy => disabled
libmongoc compression zlib => enabled

**Comment by miljojohn** [comment]
Now its working . Thank you.
but now i got error **Cannot create SSL client. SSL is not enabled in this build.","class":"MongoDB\Driver\Exception\InvalidArgumentException** 
How can i figure it out

**Comment by alcaeus** [comment]
That is a different issue that we're already discussing in https://github.com/doctrine/DoctrineMongoDBBundle/issues/452. Please let's keep the discussion in one place. Thanks.

**Pull request merged**